__tests__/app.test.js

```diff
@@ -165,4 +162,53 @@ describe("news-api", () => {
             })
         })  
     })
+    describe("POST requests", () => {
+        describe("POST /api/articles/:article_id/comments", () => {
+            test("responds with status code 201 and the posted object in expected format", () => {
+                return request(app)
+                .post("/api/articles/1/comments")
+                .send({"username": "butter_bridge", "body": "test"})
+                .expect(201)
+                .then(({body}) => {
+                    const { comment } = body
+                    expect(comment.comment_id).toEqual(expect.any(Number))
+                    expect(comment.votes).toEqual(expect.any(Number))
+                    expect(comment.created_at).toEqual(expect.any(String))
+                    expect(comment.author).toEqual(expect.any(String))
+                    expect(comment.body).toEqual(expect.any(String))
+                    expect(comment.article_id).toEqual(expect.any(Number))
+                })
+            })
+            test("responds with status code 404 'not found' if no article found with article_id", () => {
+                return request(app)
+                .post("/api/articles/9999/comments")
+                .send({"username": "rogersop", "body": "test3"})
+                .expect(404)
+                .then(({body}) => {
+                    const { msg } = body
+                    expect(msg).toBe("not found")
+                })
+            })
+            test("responds with status code 400 'bad request' when provided an article_id that is not a number", () => {
+                return request(app)
+                .post("/api/articles/not-a-number/comments")
+                .send({"username": "lurker", "body": "test4"})
+                .expect(400)
+                .then(({body}) => {
+                    const { msg } = body
+                    expect(msg).toBe("bad request")
+                })
+            })
+            test("responds with status code 422 'unprocessable entity' when provided a username that doesn't exist", () => {
+                return request(app)
+                .post("/api/articles/1/comments")
+                .send({"username": "not-a-user", "body": "test5"})
+                .expect(422)
+                .then(({body}) => {
+                    const { msg } = body
+                    expect(msg).toBe("unprocessable entity")
+                })
+            })
+        })
+    })
```

Comment:
What if the request is not complete? As in - what if we have a username but no body, or vice-versa?